libs/velo-external-db-core/src/service/data.ts

```diff
@@ -9,13 +9,14 @@ export default class DataService {
         this.storage = storage
     }
 
-    async find(collectionName: string, _filter: Filter, sort: any, skip: any, limit: any, projection: any) {
-        const items = this.storage.find(collectionName, _filter, sort, skip, limit, projection)
-        const totalCount = this.storage.count(collectionName, _filter)
+    async find(collectionName: string, _filter: Filter, sort: any, skip: any, limit: any, projection: any, omitTotalCount?: boolean): Promise<{items: any[], totalCount?: number}> {
+        const items = await this.storage.find(collectionName, _filter, sort, skip, limit, projection)
+        const totalCount = omitTotalCount? undefined : await this.storage.count(collectionName, _filter)
+
         return {
-            items: (await items).map(item => asWixData(item, projection)),
-            totalCount: await totalCount
-        }
+            items: items.map(item => asWixData(item, projection)),
+            totalCount: totalCount
+        }     
```

Comment:
will be better to put the await in the return statement, to allow the two calls run simultaneously